src/main/java/com/example/planit/controller/CalendarController.java

```diff
@@ -105,19 +105,23 @@ public ResponseEntity<DTOstatus> generateStudyEvents(@RequestParam String sub, @
     }
 
     @GetMapping(value = "/study-plan")
-    public ResponseEntity<DTOstudyPlanResponseToClient> getLatestStudyPlan(@RequestParam String sub) {
+    public ResponseEntity<DTOstudyPlanAndUpcomingSessionResponseToClient> getLatestStudyPlan(@RequestParam String sub) {
 
         long s = System.currentTimeMillis();
         logger.info(MessageFormat.format("User {0}: has requested POST /study-plan with params: sub={0}", sub));
 
-        DTOstudyPlanResponseToController dtOstudyPlanResponseToController = calendarEngine.getUserLatestStudyPlan(sub);
+        DTOstudyPlanAndUpcomingSessionResponseToController dtOstudyPlanResponseToController = calendarEngine.getUserLatestStudyPlan(sub);
+
+        DTOstudyPlanAndUpcomingSessionResponseToController dtOupcomingSessionResponseToController = calendarEngine.getUserUpcomingSession(sub);
 
```

Comment:
Hy Oshri Great Work so far !
Small comment here i would like that if you decided to change the name of the DTO to contain both studyPlan and studySession for client then i want to see a single call from controller to engine.